oxidation/libparsec/crates/core_fs/src/storage/manifest_storage.rs

```diff
@@ -485,56 +521,58 @@ impl ManifestStorage {
         }
         // Flush the cached value to the localdb
         if !cache_only {
-            self.ensure_manifest_persistent_unlock(entry_id, &mut cache_unlock)?;
+            self.ensure_manifest_persistent_unlock(entry_id, cache_unlock)
+                .await?;
         }
 
         Ok(())
     }
 
-    pub fn ensure_manifest_persistent(&self, entry_id: EntryID) -> FSResult<()> {
-        if let Some(cache_entry) = self
-            .caches
-            .lock()
-            .expect("Mutex is poisoned")
-            .get(&entry_id)
-            .cloned()
-        {
+    pub async fn ensure_manifest_persistent(&self, entry_id: EntryID) -> FSResult<()> {
+        let cache_entry = {
+            let cache_guard = self.caches.lock().expect("Mutex is poisoned");
+
+            cache_guard.get(&entry_id).cloned()
+        };
+
+        if let Some(cache_entry) = cache_entry {
             self.ensure_manifest_persistent_lock(entry_id, cache_entry)
+                .await
         } else {
             Ok(())
         }
     }
 
-    fn ensure_manifest_persistent_lock(
+    async fn ensure_manifest_persistent_lock(
         &self,
         entry_id: EntryID,
-        cache_lock: Arc<Mutex<CacheEntry>>,
+        cache_lock: Arc<AsyncMutex<CacheEntry>>,
     ) -> FSResult<()> {
-        self.ensure_manifest_persistent_unlock(
-            entry_id,
-            &mut cache_lock.lock().expect("Mutex is poisoned"),
-        )
+        self.ensure_manifest_persistent_unlock(entry_id, cache_lock.lock().await)
+            .await
     }
 
-    fn ensure_manifest_persistent_unlock(
+    async fn ensure_manifest_persistent_unlock<'a>(
         &self,
         entry_id: EntryID,
-        cache: &mut CacheEntry,
+        mut cache: AsyncMutexGuard<'a, CacheEntry>,
     ) -> FSResult<()> {
-        if let (Some(manifest), Some(ref pending_chunk_ids)) =
-            (&mut cache.manifest, &mut cache.pending_chunk_ids)
+        if let (Some(manifest), Some(pending_chunk_ids)) =
+            (cache.manifest.clone(), cache.pending_chunk_ids.clone())
         {
             let ciphered = manifest.dump_and_encrypt(&self.local_symkey);
-            let pending_chunk_ids = pending_chunk_ids
+            let chunk_ids = pending_chunk_ids
                 .iter()
-                .map(|chunk_id| chunk_id.as_ref())
+                .map(|chunk_id| chunk_id.as_ref().to_vec())
```

Comment:
Why allocate ?

Reply:
The value is used when we delete the chunks from the table.
And rust require that the value live for long enough

Reply:
That the requirement for sending value across thread, it require the `'static` lifetime :/